src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -428,22 +385,16 @@ private static bool NeedsChecked(TypeSymbol source, TypeSymbol target)
                 return false;
             }
 
-            if (source.IsDynamic())
-            {
-                return true;
-            }
-
-            SpecialType sourceST = source.StrippedType().EnumUnderlyingType().SpecialType;
-            SpecialType targetST = target.StrippedType().EnumUnderlyingType().SpecialType;
-
             // integral to double or float is never checked, but float/double to integral 
             // may be checked.
-            bool sourceIsNumeric = SpecialType.System_Char <= sourceST && sourceST <= SpecialType.System_Double;
-            bool targetIsNumeric = SpecialType.System_Char <= targetST && targetST <= SpecialType.System_UInt64;
-            return
-                sourceIsNumeric && target.IsPointerType() ||
-                targetIsNumeric && source.IsPointerType() ||
-                sourceIsNumeric && targetIsNumeric && s_needsChecked[sourceST - SpecialType.System_Char, targetST - SpecialType.System_Char];
+            return IsUnderlyingTypeInRange(source, SpecialType.System_Char, SpecialType.System_Double) &&
+                IsUnderlyingTypeInRange(target, SpecialType.System_Char, SpecialType.System_UInt64);
+        }
```

Comment:
The table was not used for Expression lambas. See the previous code at `LocalRewriter_Conversions.cs` line 111, `@checked` depended on `DistinctSpecialTypes` only in that case. (And `DistinctSpecialTypes` had overlap with this method, so the two were combined.)